src/isl/util.rs

```diff
@@ -27,6 +29,224 @@ pub enum Range {
 }
 
 impl Range {
+    /// Provides a boolean value to specify whether the given value is within the range or not
+    pub fn contains(&self, value: &OwnedElement) -> IonSchemaResult<bool> {
```

Comment:
issue—from the spec (emphasis added):
> A number range includes all values of any numeric type (i.e. `float`, `int`, and `decimal`) that fall within that range mathematically. When testing a value for inclusion, the range bounds and the value that is being tested _will all be converted to `decimal` values for the sake of comparison_. `number ranges` do not include values of any other type.

...but this implementation does not convert any of the other number types to a decimal.

I think this is also a good opportunity to separate out the normalization to decimal from the actual comparison, which will help eliminate a lot of very similar code that repeats the comparison logic for each type.

----

question—given that numbers are treated identically when it comes to comparison, do you think it would be worth refactoring `Range` into `NumberRange` and `TimestampRange`?

Reply:
I can certainly normalize ranges to decimal for comparison. 

From my understanding, you are suggesting to refactor `Range` into `NumberRange` and `TimestampRange` as all number `Range`s will be converted to `Decimal` under the hood. But from the reference of `ion-schema-kotlin` [implementation](https://github.com/amzn/ion-schema-kotlin/blob/master/src/com/amazon/ionschema/internal/util/RangeIonTimestamp.kt#L53) and from above statement it does seem like all types of ranges are being converted to `Decimal`, so maybe a single method implementation could be used for all of them. Unless, there was a specific reason you suggested this?

Reply:
> From my understanding, you are suggesting to refactor Range into NumberRange and TimestampRange as all number Ranges will be converted to Decimal under the hood. But from the reference of ion-schema-kotlin implementation and from above statement it does seem like all types of ranges are being converted to Decimal, so maybe a single method implementation could be used for all of them. Unless, there was a specific reason you suggested this?

I don't have a clearly formed idea about this—I was just throwing an idea out there to see if you had an opinion on it. Don't worry about it unless it seems to you like it would make the implementation simpler.

Reply:
Sure. Thanks. I will try to restructure this for using `Decimal` comparison and see if we can change something here.